internal/controller/datadogagent/common/utils.go

```diff
@@ -49,6 +49,14 @@ func GetDefaultMetadata(owner metav1.Object, componentKind, componentName, versi
 		for key, val := range selector.MatchLabels {
 			labels[key] = val
 		}
+		// if update selector is present, use k8s instance and component as the selector
+	} else if _, ok := owner.GetAnnotations()[apicommon.UpdateSelectorAnnotationKey]; ok {
```

Comment:
nit: as long as the annotation is present with any value, it will update the selector. Should it be gated behind the value of the annotation, e.g. true ? Not sure if it's worth noting it

Reply:
Thanks, changed it to check that the value == `true`. For the question above, the default DS pod labels don't change so they shouldn't update. I'll update the test instructions to make that clearer